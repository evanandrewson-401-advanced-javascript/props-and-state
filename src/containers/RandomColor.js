import React, { Component } from 'react';

export default class RandomC255olor extends Component {
  state = {
    backgroundColor: 'rgb(0, 0, 0)'
  };

  generateRGBvalue = () => {
    return `rgb(${Math.floor(Math.random() * (256))}, ${Math.floor(Math.random() * (256))}, ${Math.floor(Math.random() * (256))}`
  }

  setColor = (rgbValue) => {
    this.setState({ backgroundColor: rgbValue })
  }

  componentDidMount() {
    setInterval(() => this.setColor(this.generateRGBvalue()), 1000);
  }

  render() {
    const { backgroundColor } = this.state;
    return (
        <div style={{ 
          backgroundColor,
          height: '100vh',
          width: '100vw' }} />
    )
  }
}