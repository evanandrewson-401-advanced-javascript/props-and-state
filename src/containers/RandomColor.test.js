import React from 'react';
import { mount } from 'enzyme';
import RandomColor from './RandomColor';

describe('RandomColor component', () => {
  it('renders RandomColor', () => {
    const wrapper = mount(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});