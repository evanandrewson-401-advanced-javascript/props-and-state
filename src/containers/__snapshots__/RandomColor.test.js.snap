// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomColor component renders RandomColor 1`] = `
<RandomC255olor>
  <div
    style={
      Object {
        "backgroundColor": "rgb(0, 0, 0)",
        "height": "100vh",
        "width": "100vw",
      }
    }
  />
</RandomC255olor>
`;
